<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ DIANE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DIANE</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">1.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DIANE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/DIANE_Input_data.html">Data upload</a>
    </li>
    <li>
      <a href="../articles/DIANE_normalize.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/DIANE_Programming_Interface.html">DIANE Programming Interface</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/OceaneCsn/DIANE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-diane-user-interface" class="hasAnchor"><a href="#section-diane-user-interface" class="anchor"></a>DIANE User Interface</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="run_app.html">run_app()</a></code> </p>
        </td>
        <td><p>Run the Shiny Application</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-pre-processing" class="hasAnchor"><a href="#section-pre-processing" class="anchor"></a>Pre-processing</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="normalize.html">normalize()</a></code> </p>
        </td>
        <td><p>Normalize raw count data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_low_counts.html">filter_low_counts()</a></code> </p>
        </td>
        <td><p>Remove low count genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aggregate_splice_variants.html">aggregate_splice_variants()</a></code> </p>
        </td>
        <td><p>Aggregate splice variants</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_locus.html">get_locus()</a></code> </p>
        </td>
        <td><p>Get gene IDs loci, without splicing information</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-differential-expression-analysis" class="hasAnchor"><a href="#section-differential-expression-analysis" class="anchor"></a>Differential expression analysis</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="estimateDEGs.html">estimateDEGs()</a></code> </p>
        </td>
        <td><p>Estimates Differentially Expression Genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="estimateDispersion.html">estimateDispersion()</a></code> </p>
        </td>
        <td><p>Estimates gene dispersions prior to differential expression testing</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-expression-based-clustering" class="hasAnchor"><a href="#section-expression-based-clustering" class="anchor"></a>Expression based clustering</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="run_coseq.html">run_coseq()</a></code> </p>
        </td>
        <td><p>Mixture Model clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_genes_in_cluster.html">get_genes_in_cluster()</a></code> </p>
        </td>
        <td><p>Get genes in a specific cluster</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fit_glm.html">fit_glm()</a></code> </p>
        </td>
        <td><p>Fits a Poisson generalized linear model to a set of genes</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-gene-ontologies-and-annotations" class="hasAnchor"><a href="#section-gene-ontologies-and-annotations" class="anchor"></a>Gene ontologies and annotations</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="enrich_go.html">enrich_go()</a></code> </p>
        </td>
        <td><p>Enriched ontologies for recognized organisms</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="enrich_go_custom.html">enrich_go_custom()</a></code> </p>
        </td>
        <td><p>Enriched ontologies for a custom organism</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_from_agi.html">convert_from_agi()</a></code> </p>
        </td>
        <td><p>For A. thaliana, converts TAIR IDs to entrez IDs, 
gene symbol or description</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_from_ensembl.html">convert_from_ensembl()</a></code> </p>
        </td>
        <td><p>For H. sapiens, converts ensembl IDs to entrez IDs, 
symbol or name</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_from_ensembl_ce.html">convert_from_ensembl_ce()</a></code> </p>
        </td>
        <td><p>For C. elegans, converts ensembl IDs to entrez IDs, 
symbol or name</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_from_ensembl_dm.html">convert_from_ensembl_dm()</a></code> </p>
        </td>
        <td><p>For D. melanogaster, converts ensembl IDs to entrez IDs, 
symbol or name</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_from_ensembl_eck12.html">convert_from_ensembl_eck12()</a></code> </p>
        </td>
        <td><p>For E coli, converts ensembl IDs to entrez IDs, 
symbol or name</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_from_ensembl_mus.html">convert_from_ensembl_mus()</a></code> </p>
        </td>
        <td><p>For M. musculus, converts ensembl IDs to entrez IDs, 
symbol or name</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_gene_information.html">get_gene_information()</a></code> </p>
        </td>
        <td><p>Get genes information</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-gene-regulatory-network-inference" class="hasAnchor"><a href="#section-gene-regulatory-network-inference" class="anchor"></a>Gene regulatory network inference</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="network_data.html">network_data()</a></code> </p>
        </td>
        <td><p>Creates data describing a gene regulatory network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="network_from_tests.html">network_from_tests()</a></code> </p>
        </td>
        <td><p>Create network from edges statistical tests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="network_inference.html">network_inference()</a></code> </p>
        </td>
        <td><p>Regulatory importances estimation via Random Forests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="network_thresholding.html">network_thresholding()</a></code> </p>
        </td>
        <td><p>Thresholds a non sparse adjascency matrix to return the strongest weights only</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="test_edges.html">test_edges()</a></code> </p>
        </td>
        <td><p>Statistical edges testing by permutations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="estimate_test_edges_time.html">estimate_test_edges_time()</a></code> </p>
        </td>
        <td><p>Estimates time spent for statistical testing</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="group_regressors.html">group_regressors()</a></code> </p>
        </td>
        <td><p>Group correlated regulators</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="describe_node.html">describe_node()</a></code> </p>
        </td>
        <td><p>Targets and regulators of a gene in a network</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-visualization-functions" class="hasAnchor"><a href="#section-visualization-functions" class="anchor"></a>Visualization functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="draw_coseq_run.html">draw_coseq_run()</a></code> </p>
        </td>
        <td><p>Displays clustering results and quality</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_DEGs.html">draw_DEGs()</a></code> </p>
        </td>
        <td><p>MA or volcano plot for differential expression results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_discarded_edges.html">draw_discarded_edges()</a></code> </p>
        </td>
        <td><p>Draw network with removed edges in red</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_distributions.html">draw_distributions()</a></code> </p>
        </td>
        <td><p>Draw distributions of expression data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_enrich_go.html">draw_enrich_go()</a></code> </p>
        </td>
        <td><p>Plot the enriched go terms of an enrich_go result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_enrich_go_map.html">draw_enrich_go_map()</a></code> </p>
        </td>
        <td><p>Draw GO enrichment map</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_expression_levels.html">draw_expression_levels()</a></code> </p>
        </td>
        <td><p>Draw gene expression levels</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_glm.html">draw_glm()</a></code> </p>
        </td>
        <td><p>Plots the coefficients value of a Poisson generalized linear model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_heatmap.html">draw_heatmap()</a></code> </p>
        </td>
        <td><p>Draw expression heatmap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_network.html">draw_network()</a></code> </p>
        </td>
        <td><p>Interactive network view</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_network_degrees.html">draw_network_degrees()</a></code> </p>
        </td>
        <td><p>Draw a network's degree distributions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_PCA.html">draw_PCA()</a></code> </p>
        </td>
        <td><p>#' Multi-dimensional scaling plot
#'
#' @param normalized.count data to plot for MDS, with samples as columns and genes as rows
#' @importFrom limma plotMDS
#' @export
#' @examples
#' data("abiotic_stresses")
#' DIANE::draw_MDS(abiotic_stresses$normalized_counts)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_profiles.html">draw_profiles()</a></code> </p>
        </td>
        <td><p>Draw expression profiles of a clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="draw_venn.html">draw_venn()</a></code> </p>
        </td>
        <td><p>Venn diagram</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-data" class="hasAnchor"><a href="#section-data" class="anchor"></a>Data</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="regulators_per_organism.html">regulators_per_organism</a></code> </p>
        </td>
        <td><p>List of regulator genes for a number of model organisms</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="abiotic_stresses.html">abiotic_stresses</a></code> </p>
        </td>
        <td><p>Combined abiotic stresses</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gene_annotations.html">gene_annotations</a></code> </p>
        </td>
        <td><p>Gene annotations for supported organisms in DIANE</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Oceane Cassan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


